static/js/components/AddToListDialog.js

```diff
@@ -0,0 +1,271 @@
+            resource.object_type
+          ) ||
+            resource.id !== userList.id) ? (
+              <div className="flex-row" key={i}>
+                <div>
+                  <Checkbox
+                    checked={inLists.includes(userList.id)}
+                    onChange={(e: any) => {
+                      toggleListItem(resource, userList, !e.target.checked)
+                    }}
+                  >
+                    {`${userList.title}`}
+                  </Checkbox>
+                </div>
+                <div>
+                  <div className="grey-surround privacy">
+                    <i
+                      className={`material-icons ${privacyIcon(
+                        userList.privacy_level
+                      )}`}
+                    >
+                      {privacyIcon(userList.privacy_level)}
+                    </i>
+                    {capitalize(userList.privacy_level)}
+                  </div>
+                </div>
+              </div>
+            ) : null
+      )}
+    </div>
+  )
+  return resource && isFinished ? (
+    <Dialog
+      id="list-add-dialog"
+      open={!emptyOrNil(resource)}
+      hideDialog={hideDialog}
+      title="Add to List"
+      onAccept={hideDialog}
+      hideCancel={true}
+      submitText="OK"
+      className="user-listitem-dialog"
+    >
+      {renderAddToListForm()}
+    </Dialog>
+  ) : null
+}
+
+const getObject = createSelector(
+  state => state.ui,
+  state => state.entities.courses,
+  state => state.entities.bootcamps,
+  state => state.entities.programs,
+  state => state.entities.userLists,
+  state => state.entities.videos,
+  state => state.queries,
+  (ui, courses, bootcamps, programs, userLists, videos, queries) => {
+    const object = ui.dialogs.get(DIALOG_ADD_TO_LIST)
+
+    if (object && object.object_type) {
+      switch (object.object_type) {
+      case LR_TYPE_COURSE:
+        return querySelectors.isFinished(queries, courseRequest(object.id))
+          ? courses[object.id]
+          : null
+      case LR_TYPE_BOOTCAMP:
+        return querySelectors.isFinished(queries, bootcampRequest(object.id))
+          ? bootcamps[object.id]
+          : null
+      case LR_TYPE_PROGRAM:
+        return querySelectors.isFinished(queries, programRequest(object.id))
+          ? programs[object.id]
+          : null
+      case LR_TYPE_VIDEO:
+        return querySelectors.isFinished(queries, videoRequest(object.id))
+          ? videos[object.id]
+          : null
+      case LR_TYPE_USERLIST:
+        return querySelectors.isFinished(queries, userListRequest(object.id))
+          ? userLists[object.id]
+          : null
+      case LR_TYPE_LEARNINGPATH:
+        return querySelectors.isFinished(queries, userListRequest(object.id))
+          ? userLists[object.id]
+          : null
+      }
+    }
+    return null
+  }
+)
+
+export const mapStateToProps = (state: Object) => {
```

Comment:
Not sure how that would work when there is a single `AddToListDialog` in `CourseIndexPage` and `CourseSearchPage`? 

Can't set `object` as a property like this anymore:
```javascript
      <AddToListDialog
        object={object}
        hide={() => setShowAddToListDialog(false)}
      />
```
and if I do this:
```javascript
const [showAddToListDialog, setShowAddToListDialog] = useState(false)
```
then wouldn't I have to pass `setShowAddToListDialog` to `LearningResourceCard` as well since that's where the dialog is triggered?  And this needs to be done on both pages, whereas right now all the logic for hiding/showing the dialog and retrieving the correct object is self-contained in the dialog.

Reply:
sorry - wasn't clear, I was just saying it would be good to use `useDispatch` and `useSelector` instead of `connect`, `connectRequest`, etc. I think we should use only these on all new code going forward.